crates/polars-compute/src/filter/mod.rs

```diff
@@ -26,6 +26,20 @@ pub fn filter(array: &dyn Array, mask: &BooleanArray) -> Box<dyn Array> {
 }
 
 pub fn filter_with_bitmap(array: &dyn Array, mask: &Bitmap) -> Box<dyn Array> {
+    // Many filters involve filtering values in a subsection of the array. When we trim the leading
+    // and trailing filtered items, we can close in on those items and not have to perform and
+    // thinking about those. The overhead for when there are no leading or trailing filtered values
+    // is very minimal: only a clone of the mask and the array.
+    //
+    // This also allows dispatching to the fast paths way, way, way more often.
+    let mut mask = mask.clone();
```

Comment:
Maybe we should do this one line 51? After the full and empty check.

Reply:
No the effectiveness stems exactly from doing it before. If we trim to an exact full or empty block, we can take the fast path.

Reply:
Ah, right. :+1: